### PR TITLE
Events Controllers, Models, and Routes

### PR DESCRIPTION
**Pull request by nelliott82**
- Controller/Model/Routes for all events GET
- Controller/Model/Routes for one event GET

### CODE REVIEW

**Comment by nelliott82** [comment]
[Events Controllers and Models](https://trello.com/c/e9IQCdsC/31-events-controllers-and-models)

File: database/models/fans.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+              json_build_object(
+                'event_id', e.id,
+                'event_name', e.name,
+                'event_street', e.street,
+                'event_city', e.city,
+                'event_state', e.state,
+                'event_longitude', e.longitude,
+                'event_latitude', e.latitude,
+                'event_timestamp', e.timestamp,
+                'event_start', e.start,
```
Comment (ykdong) [inline database/models/fans.js:29]: 'start' at line 29 and 'end' at line 30 is named 'start_time' and 'end_time' in event table

File: database/models/fans.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        'artists', (SELECT
+          array_to_json(
+            array_agg(
+              json_build_object(
+                'artist_id', a.id,
+                'artist_display-name', a.display-name,
+                'artist_instrument', a.instrument,
+                'artist_genre', a.genre,
+                'artist_bio', a.bio,
+                'artist_picture', a.picture,
```
Comment (ykdong) [inline database/models/fans.js:48]: picture is named as 'pic' in artist table

File: database/models/fans.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                'artist_paymentmethod', a.paymentmethod,
+                'artist_fans_count', a.fans_count
```
Comment (ykdong) [inline database/models/fans.js:50]: fans_count is named as 'fan_num' in artist table

File: database/models/fans.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    })
+    .catch((err) => {
+      res.status(500);
+      console.log('GET Fan Info error: ', err);
+      res.end(JSON.stringify(err));
+    });
+  },
+  saveEvent: function(req, res) {
+    client.query(`INSERT INTO event_fan
+                  (fans_id, events_id)
```
Comment (ykdong) [inline database/models/fans.js:75]: (fans_id, events_id) are named as (fan_id, event_id) in event_fan table

File: database/models/fans.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+                  (${req.body.id}, ${req.body.event_id})`)
+    .then(() => res.sendStatus(201);)
+    .catch(err => {
+      console.log('Save Event Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  followArtist: function(req, res) {
+    client.query(`INSERT INTO artist_fan
```
Comment (ykdong) [inline database/models/fans.js:86]: artist_fan table is named as 'art_fan' in database / (fans_id, artists_id) are named as (art_id, fan_id) in art_fan table

File: database/models/fans.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    .then(() => res.sendStatus(204);)
+    .catch(err => {
+      console.log('Update profile Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  removeEvent: function(req, res) {
+    client.query(`DELETE FROM event_fan
+                  WHERE fans_id = ${req.params.fanId}
```
Comment (ykdong) [inline database/models/fans.js:114]: line 112, 113 are named as 'fan_id' , 'event_id' in event_fan table

File: database/models/fans.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    .then(() => res.sendStatus(201);)
+    .catch(err => {
+      console.log('Remove event Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  unfollowArtist: function(req, res) {
+    client.query(`DELETE FROM artist_fan
+                  WHERE fans_id = ${req.params.fanId}
```
Comment (ykdong) [inline database/models/fans.js:125]: artist_fan table is named as 'art_fan' in database / (fans_id, artists_id) are named as (art_id, fan_id) in art_fan table

File: database/models/fans.js [hunk 8]
```diff
@@ -0,0 +1,134 @@
+const client = require('../index');
+
+module.exports = {
+  get: function(req, res) {
+    client.query(`
+      SELECT json_build_object(
+       'username', (SELECT username FROM authentication a WHERE a.id = f.authentication_id),
```
Comment (ykdong) [inline database/models/fans.js:7]: 'authentication' table named as 'auth' in database,
'authentication_id' named as 'auth_id' in fans table

**Review by ykdong — APPROVED** [review]

**Pull request merged**